netsim/utils/log.py

```diff
@@ -315,9 +315,13 @@ def warning(*,
   if flag_value is False:
     return
 
+  if 'category' not in kwargs:
```

Comment:
This change seems unrelated to gateway processing... ;)

Reply:
OK, I can make it into a separate commit ;)